### ISSUE TITLE
ndcube uses deprecated "cgi" package

### ISSUE THREAD

**Issue by olebole** [sunpy/ndcube#588]
### Describe the bug

Forwarding [Debian#1030490](https://bugs.debian.org/1020490):
With Python 3.11, `collecting ndcube/utils/sphinx/code_context.py` produces an error, converted from a DeprecationWarning that the **cgi** module is now deprecated. This lets the test suite fail.

### To Reproduce

Run the tests with Python 3.11 installed.

### Screenshots

```
==================================== ERRORS ====================================
_____________ ERROR collecting ndcube/utils/sphinx/code_context.py _____________
../../../ndcube/utils/sphinx/code_context.py:8: in <module>
    from sphinx.directives.code import CodeBlock
/usr/lib/python3/dist-packages/sphinx/directives/__init__.py:13: in <module>
    from sphinx.util.docfields import DocFieldTransformer, Field, TypedField
/usr/lib/python3/dist-packages/sphinx/util/docfields.py:13: in <module>
    from sphinx.environment import BuildEnvironment
/usr/lib/python3/dist-packages/sphinx/environment/__init__.py:18: in <module>
    from sphinx.config import Config
/usr/lib/python3/dist-packages/sphinx/config.py:14: in <module>
    from sphinx.util.i18n import format_date
/usr/lib/python3/dist-packages/sphinx/util/i18n.py:11: in <module>
    from babel.messages.mofile import write_mo
/usr/lib/python3/dist-packages/babel/messages/__init__.py:11: in <module>
    from babel.messages.catalog import *
/usr/lib/python3/dist-packages/babel/messages/catalog.py:13: in <module>
    from cgi import parse_header
/usr/lib/python3.11/cgi.py:57: in <module>
    warnings._deprecated(__name__, remove=(3,13))
/usr/lib/python3.11/warnings.py:514: in _deprecated
    warn(msg, DeprecationWarning, stacklevel=3)
E   DeprecationWarning: 'cgi' is deprecated and slated for removal in Python 3.13
=========================== short test summary info ============================
```

### System Details

- Python 3.11
- ndcube 2.0.2

### Installation method

from source (.tar.gz)

**Comment by Cadair** [comment]
That import isn't in ndcube though, so we can't actually fix it? We could filter it in the warnings list if that would work for you?

**Comment by olebole** [comment]
Oh, yes, you are right! I misinterpreted that, sorry. I can filter it out locally.
Actually, turning Deprecation warnings into errors is useful mainly for developers to have an early warning (and this was the reason why I opened the issue). When packaging (like for Debian), they are not useful, since they return an error even when the package works perfectly from a user's point of view. But since I can do local patches, this is no problem for me.
IMO this could be closed then.
